lib/codecs/src/decoding/format/protobuf.rs

```diff
@@ -0,0 +1,365 @@
+use std::collections::BTreeMap;
+use std::fs;
+
+use bytes::Bytes;
+use chrono::Utc;
+use lookup::PathPrefix;
+use ordered_float::NotNan;
+use prost_reflect::{DescriptorPool, DynamicMessage, MessageDescriptor, ReflectMessage};
+use smallvec::{smallvec, SmallVec};
+use vector_config::configurable_component;
+use vector_core::event::LogEvent;
+use vector_core::{
+    config::{log_schema, DataType, LogNamespace},
+    event::Event,
+    schema,
+};
+use vrl::value::Kind;
+
+use super::Deserializer;
+
+/// Config used to build a `ProtobufDeserializer`.
+#[configurable_component]
+#[derive(Debug, Clone, Default)]
+pub struct ProtobufDeserializerConfig {
+    /// Path to desc file
+    desc_file: String,
```

Comment:
File names should be stored in `PathBuf` type instead of strings to avoid corner cases of illegal filenames.